tests/Xamarin.Android.Timing/TimingLogger.cs

```diff
@@ -0,0 +1,358 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Text;
+using System.Xml;
+
+using Microsoft.Build;
+using Microsoft.Build.Evaluation;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Xamarin.Android.Timing
+{
+	public class TimingLogger : Logger
+	{
+		class LogEvent
+		{
+			public XmlElement Element;
+			public DateTime Start;
+		}
+
+		const string DefaultOutputFileName = "xa_timing.xml";
+
+		string outputFileName;
+		string buildID;
+		string buildDescription;
+		string buildCommitHash;
+		bool overwriteOutputFile;
+
+		XmlDocument doc;
+		Stack<LogEvent> events;
+		bool haveSolutionDirectory;
+		XmlElement toolsets;
+		XmlElement files;
+		int fileID = 0;
+
+		Dictionary<string, int> fileCache = new Dictionary<string, int> (StringComparer.OrdinalIgnoreCase);
+
+		public override void Initialize (IEventSource eventSource)
+		{
+			ParseParameters (Parameters?.Split (';'));
+
+			if (String.IsNullOrEmpty (outputFileName))
+				outputFileName = DefaultOutputFileName;
+
+			if (String.IsNullOrEmpty (buildID))
+				buildID = MakeDefaultBuildId ();
+
+			eventSource.BuildStarted += OnBuildStarted;
+			eventSource.BuildFinished += OnBuildFinished;
+			eventSource.ProjectStarted += OnProjectStarted;
+			eventSource.ProjectFinished += OnProjectFinished;
+			eventSource.TargetStarted += OnTargetStarted;
+			eventSource.TargetFinished += OnTargetFinished;
+			eventSource.TaskStarted += OnTaskStarted;
+			eventSource.TaskFinished += OnTaskFinished;
+
+			events = new Stack<LogEvent> ();
+			doc = new XmlDocument ();
```

Comment:
`XmlDocument`?! I feel gross. :-/

HHOS?

Reply:
This is the only one I didn't address, I agree I would rather use `System.Xml.Linq`, too.

I just didn't think it was worth rewriting all this code--but if we were going to change the file drastically or something it might be worth it.